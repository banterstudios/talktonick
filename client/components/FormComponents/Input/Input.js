import React from 'react'

import classNames from 'utils/classNames'

import PropTypes from 'prop-types'

const Input = (props) => {
  const {
    type,
    name,
    value,
    onBlur,
    onFocus,
    onInput,
    onChange,
    disabled,
    className
  } = props

  const modClassNames = classNames('input', className)

  return (
    <div className={modClassNames}>
      <input
        className='input__elem'
        type={type}
        name={name}
        value={value}
        onBlur={onBlur}
        onFocus={onFocus}
        onInput={onInput}
        onChange={onChange}
        disabled={disabled}
        />
    </div>
  )
}

Input.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.any,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onInput: PropTypes.func,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ])
}

export default Input
