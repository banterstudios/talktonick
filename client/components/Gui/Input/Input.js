/* eslint-disable no-unused-vars */
import React from 'react'
/* eslint-enable no-unused-vars */

import PropTypes from 'prop-types'

import glamorous from 'glamorous'

const Input = glamorous.input(({ theme: {
  inputHighlight,
  inputBg,
  fontSizeText,
  fontSecondary,
  inputTextColor,
  inputPlaceholderColor,
  inputNormalBorderColor,
  formBorderRadius
} }) => ({
  border: `1px solid transparent`,
  borderRadius: formBorderRadius,
  padding: '15px',
  appearance: 'none',
  backgroundColor: inputBg,
  minHeight: '50px',
  transition: 'border .2s ease-out',
  width: '100%',
  fontFamily: fontSecondary,
  fontWeight: 600,
  fontSize: fontSizeText,
  color: inputTextColor,
  '&:focus, &:active': {
    outline: 'none',
    borderColor: inputHighlight
  },
  '::placeholder': {
    color: inputPlaceholderColor
  }
}))

Input.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.any,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onInput: PropTypes.func,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  tabIndex: PropTypes.number,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ])
}

export default Input
