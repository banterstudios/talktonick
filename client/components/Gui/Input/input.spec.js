import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Input from './Input'

describe('components/Gui/Input', () => {
  let initialProps
  
  beforeEach(() => {
    initialProps = {
      type: 'text',
      name: 'input',
      value: 'hello',
      onBlur: jest.fn(),
      onFocus: jest.fn(),
      onInput: jest.fn(),
      onChange: jest.fn(),
      disabled: false
    }
  })

  describe('@renders', () => {
    it('in default state', () => {
      const component = mount(<Input { ...initialProps} />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })

  describe('@events', () => {
    describe('clicking on', () => {
      it('onBlur will trigger a callback', () => {
        const component = mount(<Input { ...initialProps } />)
        component.simulate('blur')

        expect(initialProps.onBlur).toHaveBeenCalled()
        component.unmount()
      })

      it('onFocus will trigger a callback', () => {
        const component = mount(<Input { ...initialProps } />)
        component.simulate('focus')

        expect(initialProps.onFocus).toHaveBeenCalled()
        component.unmount()
      })

      it('onInput will trigger a callback', () => {
        const component = mount(<Input { ...initialProps } />)
        component.simulate('input')

        expect(initialProps.onInput).toHaveBeenCalled()
        component.unmount()
      })

      it('onChange will trigger a callback', () => {
        const component = mount(<Input { ...initialProps } />)
        component.simulate('change')

        expect(initialProps.onChange).toHaveBeenCalled()
        component.unmount()
      })
    })
  })
})
