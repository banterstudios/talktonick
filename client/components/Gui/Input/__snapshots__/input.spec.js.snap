// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Gui/Input @renders in default state 1`] = `
<glamorous(input)
  disabled={false}
  name="input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  type="text"
  value="hello"
>
  <input
    className="css-1v53l60"
    disabled={false}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    type="text"
    value="hello"
  />
</glamorous(input)>
`;
